content/docs/instrumenting/clientlib_guidelines.md

```diff
@@ -0,0 +1,250 @@
+
+### Naming
+
+Client libraries SHOULD follow function/method/class names mentioned in this document, keeping in mind the naming conventions of the language they’re working in. For example, `set_to_current_time()` is good for a method name Python, but `SetToCurrentTime()` is better in Go and `setToCurrentTime()` is the convention in Java. Where names differ for technical reasons (e.g. not allowing function overloading), documentation/help strings SHOULD point users towards the other names.
+
+Libraries MUST NOT offer functions/methods/classes with the same or similar names to ones given here, but with different semantics.
+
+## Standard metrics types
+
+Counter, Gauge, Summary and Histogram are the primary interface by users.
+
+Counter and Gauge MUST be part of the client library. At least one of Summary and Histogram MUST be offered.
+
+These should be primarily used as file-static variables, that is, global variables defined in the same file as the code they’re instrumenting. The client library SHOULD enable this. The common use case is instrumenting a piece of code overall, not a piece of code in the context of one instance of an object. Users shouldn’t have to worry about plumbing their metrics throughout their code, the client library should do that for them (and if it doesn’t, users will write a wrapper around the library to make it "easier" - which rarely tends to go well).
+
+There MUST be a default CollectorRegistry, the standard metrics MUST by default implicitly register into it with no special work required by the user. There MUST be a way to have metrics not register to the default CollectorRegistry, for use in batch jobs and unittests. Custom collectors SHOULD also follow this.
+
+Exactly how the metrics should be created varies by language. For some (Java, Go) a builder approach is best, whereas for others (Python) function arguments are rich enough to do it in one call.
+
+For example in the Java Simpleclient we have:
+
+```java
+class YourClass {
+  static final Counter requests = Counter.build()
+      .name("requests_total")
+      .help("Requests.").register();
+}
+```
+
+This will register requests with the default CollectorRegistry. By calling `build()` rather than `register()` the metric won’t be registered (handy for unittests), you can also pass in a CollectorRegistry to `register()` (handy for batch jobs).
+
+### Counter
+
+Counter is a monotonically increasing counter. It MUST NOT allow the value to decrease, however it MAY be reset to 0 (such as by server restart).
+
+A counter MUST have the following methods:
+
+- `inc()`: Increment the counter by 1
+- `inc(double v)`: Increment the counter by the given amount. MUST check that v >= 0.
+
+A counter is ENCOURAGED to have:
+
+A way to count exceptions throw/raised in a given piece of code, and optionally only certain types of exceptions. This is count_exceptions in Python.
+
+Counters MUST start at 0.
+
+### Gauge
+
+Gauge represents a value that can go up and down.
```

Comment:
Let's link all the metric names to the respective page on https://prometheus.io/docs/concepts/metric_types/. E.g. `[Gauge](/docs/concepts/metric_types/#gauge)`.